### PR TITLE
Make `o.p.client.http.Status` safe against unknown HTTP status codes

### PR DESCRIPTION
**Pull request by snazy**

### CODE REVIEW

File: api/client/src/main/java/org/projectnessie/client/http/Status.java [hunk 1]
```diff
@@ -15,55 +15,132 @@
...
+  public static final Status SERVICE_UNAVAILABLE =
+      addCode(SERVICE_UNAVAILABLE_CODE, "Service Unavailable");
+  public static final int GATEWAY_TIMEOUT_CODE = 504;
+  public static final Status GATEWAY_TIMEOUT = addCode(GATEWAY_TIMEOUT_CODE, "Gateway Timeout");
+  public static final int HTTP_VERSION_NOT_SUPPORTED_CODE = 505;
+  public static final Status HTTP_VERSION_NOT_SUPPORTED =
+      addCode(HTTP_VERSION_NOT_SUPPORTED_CODE, "HTTP Version Not Supported");
+  public static final int NETWORK_AUTHENTICATION_REQUIRED_CODE = 511;
+  public static final Status NETWORK_AUTHENTICATION_REQUIRED =
+      addCode(NETWORK_AUTHENTICATION_REQUIRED_CODE, "Network Authentication Required");
```
Comment (snazy) [inline api/client/src/main/java/org/projectnessie/client/http/Status.java:127]: That's all regexp search+replace "magic"

**Review by adutra — APPROVED** [review]

File: api/client/src/main/java/org/projectnessie/client/http/Status.java [hunk 2]
```diff
@@ -15,55 +15,132 @@
...
-  Status(final int statusCode, final String reason) {
+  private static Status addCode(int code, String reason) {
+    Status[][] lines = LINES;
+    int line = code / 100;
+    int code100 = code % 100;
+    Status[] l = lines[line];
+    if (l.length < code100 + 1) {
+      l = lines[line] = Arrays.copyOf(l, code100 + 1);
+    }
+    return l[code100] = new Status(code, reason);
```
Comment (dimas-b) [inline api/client/src/main/java/org/projectnessie/client/http/Status.java:140]: `n % 100 + ( ((n / 100) - 2) * 10 + ((n / 100) - 2) / 3 * 22)` gives an non-overlapping index  between `0` and `63` (incl) for known codes ;)

Comment (snazy) [reply api/client/src/main/java/org/projectnessie/client/http/Status.java:140]: ???

Comment (dimas-b) [reply api/client/src/main/java/org/projectnessie/client/http/Status.java:140]: You can use a small (one level) array with that formula for computing the index from the status code.

**Pull request merged**